docs/src/index.md

```diff
@@ -0,0 +1,457 @@
+# Introduction
+
+Metaprogramming tools for DataFrames.jl objects.
+These macros improve performance and provide more convenient syntax.
```

Comment:
maybe add 1-2 examples of DataFrames.jl code and equivalent DataFramesMeta.jl code and say that they show this convenience.

Also maybe comment in what cases the macros improve performance somewhere (probably not here)

Reply:
Added an example of the convenience of DataFramesMeta. 

I dont think any of them improve performance over DataFrames. I will defer writing this section and delete the claim at the top of the document.